source/extensions/filters/network/sni_dynamic_forward_proxy/proxy_filter.cc

```diff
@@ -33,14 +35,23 @@ Network::FilterStatus ProxyFilter::onNewConnection() {
     return Network::FilterStatus::Continue;
   }
 
-  // TODO(lizan): implement circuit breaker in SNI dynamic forward proxy like it is in HTTP:
-  // https://github.com/envoyproxy/envoy/blob/master/source/extensions/filters/http/dynamic_forward_proxy/proxy_filter.cc#L65
+  circuit_breaker_ = config_->cache().canCreateDnsRequest(absl::nullopt);
+
+  if (circuit_breaker_ == nullptr) {
```

Comment:
should we be incrementing any stats when this happens?

Reply:
`canCreateDnsRequest` increment pending requests overflow on dns cache circuit breaker manager.